src/build-tools/build-binaries.ts

```diff
@@ -0,0 +1,80 @@
+import fs from 'node:fs'
+import { mkdir, writeFile } from 'node:fs/promises'
+import path from 'node:path'
+
+import pack from '@vercel/ncc'
+import archiver from 'archiver'
+import { compile } from 'nexe'
+
+
+const targets = [
+	'linux-arm64',
+	'linux-x64',
+	'mac-arm64',
+	// The Intel Mac Build still isn't working properly yet.
+	// 'mac-x64',
+	'windows-x64',
+]
+const nodeVersion = process.version.substring(1)
+
+// This allows us to the the binary either:
+//   node dist/src/build-tools/build-binaries.js # after building of course
+//   ts-node src/build-tools/build-binaries.ts
+const workspaceDir = import.meta.dirname.endsWith('dist/src/build-tools')
+	? path.join(import.meta.dirname, '..', '..', '..')
+	: path.join(import.meta.dirname, '..', '..')
+console.log(`import.meta.dirname = ${import.meta.dirname}`)
+console.log(`workspaceDir = ${workspaceDir}`)
+const distBinDir = path.join(workspaceDir, 'dist_bin')
+await mkdir(distBinDir, { recursive: true })
+
+// First pack into a single JavaScript file using ncc.
+const { code } = await pack(
+	path.join(workspaceDir, 'dist/src/run.js'),
+	{
+		// An error occurs while minifying so it doesn't happen anyway. Skip for now.
+		// (It probably wouldn't make much difference anyway in this situation.)
+		// minify: true,
+	},
+)
+
+const packFile = path.join(distBinDir, 'smartthings.mjs')
+console.log(`Writing packed JavaScript to ${packFile}`)
+await writeFile(packFile, code)
+
+const buildAndZipTarget = async (target: string): Promise<void> => {
+	console.log(`Compiling for ${target}:`)
+	const binDir = path.join(distBinDir, target)
+	const binName = path.join(binDir, `smartthings${target.startsWith('windows') ? '.exe' : ''}`)
+	await compile({
+		name: 'smartthings',
+		input: packFile,
+		resources: ['package.json'],
+		remote: 'https://github.com/SmartThingsCommunity/cli-nexe-builds/releases/download/1.0.0',
```

Comment:
This doesn't seem to resolve?  Should the 1.0.0 be the node version?

Reply:
This is correct. It is true that it doesn't resolve in a web browser but I don't think nexe ever tries to resolve this URL by itself. It's adding the asset name to it and resolving that.

(Look at the URLs for the assets on this page:

https://github.com/SmartThingsCommunity/cli-nexe-builds/releases/tag/1.0.0

It's the base of all of them.)